src/main/java/seedu/allonus/expense/ExpenseTracker.java

```diff
@@ -96,8 +115,81 @@ private static void addExpense(ArrayList<Expense> list, Expense e) {
         Expense.setNoOfItems(Expense.getNoOfItems() + EXPENSE_INDEX);
     }
 
+    /**
+     * Edits an existing expense record using its index chosen by the user.
+     *
+     * @param list  list of expenses itself
+     * @param index the index of the item to be edited
+     * @param ui    ui object to collect user's inputs
+     * @throws IndexOutOfBoundsException if the expense record is not found
+     */
+    private static void editExpense(ArrayList<Expense> list, int index, TextUi ui) throws IndexOutOfBoundsException {
```

Comment:
Does this function throw an exception? It seems to already be caught at line 141?

Reply:
Thanks for the catch! Will edit it accordingly.